src/main/java/net/bramp/ffmpeg/builder/FFmpegHlsOutputBuilder.java

```diff
@@ -0,0 +1,123 @@
+
+        return getThis();
+    }
+
+    /**
+     * hls_segment_filename Examples <br>
+     * <br>
+     * "file%03d.ts" ->  segment files: file000.ts, file001.ts, file002.ts, etc.
+     *
+     * @param filename hls_segment_file_name to set
+     * @return {@link FFmpegHlsOutputBuilder}
+     */
+    public FFmpegHlsOutputBuilder setHlsSegmentFileName(String filename) {
+        this.hls_segment_filename = checkNotEmpty(filename, "filename must not be empty");
+
+        return getThis();
+    }
+
+    /**
+     * <strong>Segment will be cut on the next key frame after this time has passed on the first m3u8 list.</strong> <br>
+     *
+     * @param duration hls_init_time to set
+     * @return {@link FFmpegHlsOutputBuilder}
+     */
+    public FFmpegHlsOutputBuilder setHlsInitTime(long duration, TimeUnit units) {
+        checkNotNull(units);
+        this.hls_init_time = units.toMillis(duration);
+
+        return getThis();
+    }
+
+    /**
+     * <strong>Set the maximum number of playlist entries. If set to 0 the list file will contain all the segments .</strong> <br>
+     * Default value is 5 <br>
+     *
+     * @param size hls_time to set
+     * @return {@link FFmpegHlsOutputBuilder}
+     */
+    public FFmpegHlsOutputBuilder setHlsListSize(int size) {
+        checkArgument(size >= 0, "Size cannot be less than 0.");
+        this.hls_list_size = size;
+
+        return getThis();
+    }
+
+    /**
+     * <strong>Append baseurl to every entry in the playlist.
+     * Useful to generate playlists with absolute paths. <br>
+     * Note that the playlist sequence number must be unique for each segment and it is not to be confused with the segment filename sequence number which can be cyclic, for example if the wrap option is specified.</strong> <br>
+     *
+     * @param baseurl hls_base_url to set
+     * @return {@link FFmpegHlsOutputBuilder}
+     */
+    public FFmpegHlsOutputBuilder setHlsBaseUrl(String baseurl) {
+        this.hls_base_url = checkNotEmpty(baseurl, "baseurl must not be empty");
+
+        return getThis();
+    }
+
+    @Override
+    protected void addFormatArgs(ImmutableList.Builder<String> args) {
+        super.addFormatArgs(args);
+        if (hls_time != null) {
+            args.add("-hls_time", toTimecode(hls_time, TimeUnit.MILLISECONDS));
+        }
+
+        if (!Strings.isNullOrEmpty(hls_segment_filename)) {
+            args.add("-hls_segment_filename", hls_segment_filename);
+        }
+
+        if (hls_init_time != null) {
+            args.add("-hls_init_time",toTimecode(hls_init_time, TimeUnit.MILLISECONDS));
+        }
+
+        if (hls_list_size != null) {
+            args.add("-hls_list_size",hls_list_size.toString());
+        }
+
+        if (!Strings.isNullOrEmpty(hls_base_url)){
+            args.add("-hls_base_url",hls_base_url);
+        }
+    }
+
+    @CheckReturnValue
+    @Override
+    protected FFmpegHlsOutputBuilder getThis() {
```

Comment:
Why getThis(), instead of returning this?

Also this makes me realise a problem with this pattern, as soon as you call a non-HlsOutputBuilder method you lose access to the HlsOutputBuilder, e.g:

```java
new FFmpegBuilder()
    .addHlsOutput("output.m3u8")
    .setVideoBitRate(..)
    .setHlsBaseUrl(String baseurl) // This line won't work.
    .build();
```

I suspect some kind wider change with Generics would make this work.

Reply:
ah yes, I think we may want to change to something like this: https://stackoverflow.com/a/7512970/88646

Reply:
Oh, there was such a problem.
As you recommended, I will try to modify it using generic.